keystore/src/connection/platform/generic/mod.rs

```diff
@@ -27,8 +29,28 @@ pub struct SqlCipherConnection {
     path: String,
 }
 
+pub(crate) struct TransactionWrapper<'conn> {
+    transaction: Transaction<'conn>,
+}
+impl<'conn> TransactionWrapper<'conn> {
+    // this is async just to conform with the wasm impl
+    pub(crate) async fn commit_tx(self) -> CryptoKeystoreResult<()> {
+        Ok(self.transaction.commit()?)
+    }
+}
+
+impl<'conn> Deref for TransactionWrapper<'conn> {
+    type Target = Transaction<'conn>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.transaction
+    }
+}
+
 unsafe impl Send for SqlCipherConnection {}
 unsafe impl Sync for SqlCipherConnection {}
+unsafe impl<'conn> Send for TransactionWrapper<'conn> {}
```

Comment:
I don't like this, but since we're using the transaction in a single operation and afterwards dropping it, I don't think it should be a problem